internal/js/modules/k6/browser/tests/locator_test.go

```diff
@@ -1172,6 +1172,180 @@ func TestLocatorFilter(t *testing.T) {
 	})
 }
 
+func TestFrameLocatorLocatorOptions(t *testing.T) {
+	t.Parallel()
+
+	// We'll only test nil and non-nil LocatorOptions here, as the actual
+	// filtering logic is tested in TestLocatorLocatorOptions. This test
+	// just ensures that FrameLocator.Locator passes the options down correctly.
+
+	setup := func(t *testing.T) *common.FrameLocator {
+		t.Helper()
+
+		tb := newTestBrowser(t)
+		p := tb.NewPage(nil)
+		err := p.SetContent(`
+			<iframe srcdoc='
+				<section>
+					<div>
+						<span>hello</span>
+					</div>
+					<div>
+						<span>world</span>
+					</div>
+				</section>
+			'></iframe>`,
+			nil,
+		)
+		require.NoError(t, err)
+		return p.Locator("iframe", nil).ContentFrame()
+	}
+	t.Run("nil_options", func(t *testing.T) {
+		t.Parallel()
+
+		n, err := setup(t).
+			Locator("div", nil).
+			Count()
+		require.NoError(t, err)
+		require.Equal(t, 2, n)
+	})
+	t.Run("options", func(t *testing.T) {
+		t.Parallel()
+
+		n, err := setup(t).
+			Locator("div", &common.LocatorOptions{
+				HasText: "hello",
+			}).
+			Count()
+		require.NoError(t, err)
+		require.Equal(t, 1, n)
+	})
+}
+
+func TestLocatorLocatorOptions(t *testing.T) {
```

Comment:
This would be maybe a bit more concise using an array of test cases.

Reply:
I thought about that, but it becomes an unreadable mess :(